lib/id3tag/frames/v2/chapter_frame.rb

```diff
@@ -0,0 +1,48 @@
+module ID3Tag
+  module Frames
+    module V2
+      class ChapterFrame < BasicFrame
+        def element_id
+          @element_id ||= StringUtil.split_by_null_byte(raw_content).first
+        end
+
+        def start_time
+          @start_time ||= parts[:start_time].unpack1("N")
```

Comment:
Leaned something new today - I did not know such unpack1 exists :)

Do you think it would be possible and beneficial to avoid storing various string in instance variables and use single cached hash `@parts`? All the spitting and unpacking should happen in single method though.

Reply:
> I did not know such unpack1 exists :)
This was a prompt from standardrb, after writing `.unpack("N")[0]`

> Do you think it would be possible and beneficial to avoid storing various string in instance variables and use single cached hash @parts? All the spitting and unpacking should happen in single method though.

Do you mean something like:

```rb
def start_time
  parts[:start_time]
end
…
def read_parts!
  …
  parts[:start_time] = unpack_number(raw_content_io.read(4))
  …
end

def unpack_number(string)
  string.unpack1("N")
end
```

Reply:
Not really. Here is an example how it could be done in table of contents frame:

```ruby

        def element_id
          unpacked[:element_id]
        end

        def top_level?
          unpacked[:flags] & FLAG_A_TOP_LEVEL_BIT > 0
        end

        def ordered?
          unpacked[:flags] & FLAG_B_ORDERED_BIT > 0
        end

        def entry_count
          unpacked[:entry_count]
        end

        def child_element_ids
          unpacked[:child_element_ids]
        end

        private

        def unpacked
          return @unpacked if defined?(@unpacked)
          @unpacked = {}
          raw_content_io.rewind
          @unpacked[:element_id] = ID3Tag::IOUtil.read_until_terminator(raw_content_io, 1)
          @unpacked[:flags] = raw_content_io.read(1).unpack1("C")
          @unpacked[:entry_count] = raw_content_io.read(1).unpack1("C")
          @unpacked[:child_element_ids] = StringUtil.split_by_null_bytes(raw_content_io.read)
          @unpacked
        end
```
This way all the work with IO object is done once and in single place - no need to reset/seek IO object position in various methods.

Reply:
👍 Done in e2ee5bf48676d7aa6ac1b3de3d739dd1d46d2771